src/main/java/org/kairosdb/core/reporting/Tagger.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2020 Dropbox
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kairosdb.core.reporting;
+
+import com.google.common.collect.SetMultimap;
+
+import java.util.function.Supplier;
+
+public interface Tagger {
+
+    /**
+     * Applies the tags from the available information to the
```

Comment:
Can you add another sentence here elaborating on the usage? It isn't immediately clear from the interface what `metricName` is supposed to be used for or if it's expected to sometimes go unused. I expect this is a result of unifying the metric name tagging (`MetricNameTagger`) and mapped tags (`TagTagger`) under the same interface, but I'd rather be explicit in the documentation.

Reply:
How's that?